var/spack/repos/builtin/packages/py-torch/package.py

```diff
@@ -247,8 +247,9 @@ class PyTorch(PythonPackage, CudaPackage, ROCmPackage):
         depends_on("cuda@9.2:11.4", when="@1.6:1.9+cuda")
         depends_on("cuda@9:11.4", when="@:1.5+cuda")
     # https://github.com/pytorch/pytorch#prerequisites
-    depends_on("cudnn@8.5:", when="@2.3:+cudnn")
-    depends_on("cudnn@7:", when="@1.6:+cudnn")
+    # https://github.com/pytorch/pytorch/issues/119400
+    depends_on("cudnn@8.5:9.0", when="@2.3:+cudnn")
+    depends_on("cudnn@7:8", when="@1.6:2.2+cudnn")
```

Comment:
@adamjstewart perhaps this needs an update for the newer versions of py-torch [and related pkgs?]

https://github.com/pytorch/pytorch?tab=readme-ov-file#prerequisites says:

```
[NVIDIA cuDNN](https://developer.nvidia.com/cudnn) v8.5 or above
```
so presumably `cudnn@9:` might work here? [at-least for the newer py-torch versions]

I noticed this when building py-torch along with py-jax - where py-jax was picking up cudnn@9.2 while py-torch is picking up cudnn@8.9

Reply:
According to https://github.com/pytorch/pytorch/issues/119400, CUDNN 9.1+ is still not yet supported. Have you tried building with 9.1+?

Reply:
Ah ok. No haven't tried using 9.1+with latest pytorch .

Since this change was made a few months ago - I assumed the newer versions of pytorch [that got released subsequently] might have had the fix. It wasn't clear to me  [from the above pytorch readme] if it got fixed or not - hence this check.